pkg/clustermanager/cluster_manager.go

```diff
@@ -829,6 +831,13 @@ func (c *ClusterManager) InstallCustomComponents(ctx context.Context, clusterSpe
 	if err := c.clusterClient.installCustomComponents(ctx, clusterSpec, cluster); err != nil {
 		return err
 	}
+	fullLifecycleAPI := features.IsActive(features.FullLifecycleAPI())
+	if fullLifecycleAPI {
+		err := c.clusterClient.SetEksaControllerEnvVar(ctx, features.FullLifecycleAPIEnvVar, strconv.FormatBool(fullLifecycleAPI), cluster.KubeconfigFile)
```

Comment:
I think automating this is awesome, thanks a lot!

However, I would vote to use flags for the controller instead of env vars
Main reason: env vars are mutable and flags are not. Changing the flags always gets reflected in the deployment spec. But env vars can be changed withing the pod.

I think env vars for the cli is not a bad idea, since it is not supposed to run for long. But the controllers is supposed to be a long running "process"

Reply:
I agree, but the only issue with that is being able to set the flag when running the binary, where we don't support being able to edit the controller flags right now through kustomize or templating. We have some precedence in using flags so I would suggest going with this approach and creating an item to support using flags instead in the lifecycle story where we can modify it when deploying the controller. Do you have any other suggestions that we can implement immediately in this PR?

Reply:
We could do something similar to the envar and just update the deployment container.command on the fly
I would vote to give that a try. If it turns out to be more complicated that what it sounds, then I'm good with addressing separately in within this project

Reply:
I looked into this and it seems like it doesn't seem to be worth to use the patch commands or get the entire yaml and figure out where to inject the arg if we are anyways going to replace it with the approach of deploying the controller with the flag set through kustomize or templating. Going to go with the current approach.